File: lab07/identify_artist.pl
#!/usr/bin/perl
foreach $text (@ARGV){   #put all files in an array ???????
        %hash1=();
	%hash2=();
	%copy=();
	open DATA, '<',$text or die;
	@data=<DATA>;
	close DATA;
	foreach $sent (@data){
		$sent = lc $sent;
		@content= $sent =~ /[A-Za-z]+/g;
		foreach $word(@content){
			if(exists $hash1{$word}){
				$hash1{$word} += 1;
			}else {
				$hash1{$word} = 1;
				$copy{$word} = 0;
			}	
		}
	}
	foreach $file (glob "lyrics/*.txt") {
		%copy=();
        	my $total=0;
        	open FILE ,'<',$file or die;
        	@files=<FILE>;
        	close FILE;
        	$file =~ s/lyrics\///g;
        	$file =~ s/.txt//g;
        	$file =~ s/_/ /g;
        	foreach $line (@files) {
                	$line = lc $line;
                	@all = $line =~ /[A-Za-z]+/g;
			$total += $#all +1;
			foreach $key (keys %hash1){
                		@array = $line =~ /\b$key\b/g;   #match a word boundary
                		$copy{$key} += $#array+1;
			}
        	}
#foreach $key (keys %hash1){print "$file for key $key is $copy{$key}\n"};
		$result = 0;
		foreach $key (keys %copy){$result += log(($copy{$key}+1)/$total)*$hash1{$key};}
		$hash2{$file} = $result;
	}
	
	$name = (sort { $hash2{$b} <=> $hash2{$a} } keys %hash2)[0];
	$log = sprintf("%.1f",$hash2{$name});
	print "$text most resembles the work of $name (log-probability=$log)\n";
}

File: lab07/identify_artist.py
#!/usr/bin/python
import re, sys, glob,math


for file in sys.argv[1:]:
	content = open(file,"r")
	h = dict()
	copy= dict()
	singer = dict()

	for line in content:
		line = line.lower()
		line = line.rstrip()
	  	for word in re.split(r'[^a-z]+',line):
			if word !="":
				if word in h:
					h[word] += 1
				else:
					h[word] = 1
					copy[word] = 0
	#for x in h:
	#	print "For "+ x + " is "+ str(h[x])
	for musicfile in sorted(glob.glob("lyrics/*.txt")):
		total = 0
		for key in copy:
			copy[key]=0
	    	content = open(musicfile,"r")
	    	name = re.sub("lyrics/","",musicfile)
	    	name = re.sub(".txt","",name)
	    	name = re.sub("_"," ",name)
		for line in content:
			line = line.lower()
			all = re.findall("[A-Za-z]+",line)
			total += len(all)

			for key in h:
				reg= r"\b"+re.escape(key)+r"\b"
				array = re.findall(reg,line)
				copy[key] += len(array)

		result = 0

		for key in h:
			result += h[key]*math.log(float(copy[key]+1)/float(total))

		singer[name] = result

#	for w in sorted(singer, key=singer.get,reverse=True):   here is a way sort the value
	val = max(singer, key=singer.get)
	print file + " most resembles the work of "+val+ "(log-probability="+"{:.1f}".format(singer[val])+")"
